### PR TITLE
chore: list command namespace

### PR DESCRIPTION
**Pull request by lkingland**
- Adds namespace test to list command
- Moves namespace flag into list command for
  - Better help text
  - Easier testing
- Sets static default namespace for flag to "default"

/kind techdebt

### CODE REVIEW

File: cmd/list_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+	fn "knative.dev/func"
+	"knative.dev/func/mock"
+)
+
+// TestList_Namespace ensures that list command options for specifying a
+// namespace (--namespace) or all namespaces (--all-namespacs) are  properly
+// evaluated.
+func TestList_Namespace(t *testing.T) {
+	_ = fromTempDirectory(t)
```
Comment (matejvasek) [inline cmd/list_test.go:14]:
```suggestion
func TestList_Namespace(t *testing.T) {
	t.Setenv("KUBECONFIG", "/tmp/non-existent")
	_ = fromTempDirectory(t)
```

Comment (matejvasek) [reply cmd/list_test.go:14]: In order to not interfere with developers config.

Comment (lkingland) [reply cmd/list_test.go:14]: Ignoring KUBECONFIG of current user for unit tests now being tracked in a separate PR

Comment (matejvasek) [reply cmd/list_test.go:14]: What PR @lkingland ? Has it been fixed?

**Comment by matejvasek** [comment]
@lkingland unit test running with current `main` merged in are failing.

**Comment by matejvasek** [comment]
Assert need adjustment and/or you need also rebase.

**Comment by matejvasek** [comment]
@lance can you override the coverage thing? It's obviously broken.

**Pull request merged**